includes/collection/class-actors.php

```diff
@@ -0,0 +1,290 @@
+<?php
+/**
+ * Actors collection file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Collection;
+
+use WP_Error;
+use WP_User_Query;
+use Activitypub\Model\User;
+use Activitypub\Model\Blog;
+use Activitypub\Model\Application;
+
+use function Activitypub\object_to_uri;
+use function Activitypub\normalize_url;
+use function Activitypub\normalize_host;
+use function Activitypub\url_to_authorid;
+use function Activitypub\is_user_disabled;
+
+/**
+ * Actors collection.
+ */
+class Actors {
+	/**
+	 * The ID of the Blog Actor.
+	 *
+	 * @var int
+	 */
+	const BLOG_USER_ID = 0;
+
+	/**
+	 * The ID of the Application Actor.
+	 *
+	 * @var int
+	 */
+	const APPLICATION_USER_ID = -1;
+
+	/**
+	 * Get the Actor by ID.
+	 *
+	 * @param int $user_id The User-ID.
+	 *
+	 * @return User|Blog|Application|WP_Error The Actor or WP_Error if user not found.
+	 */
+	public static function get_by_id( $user_id ) {
+		if ( is_string( $user_id ) || is_numeric( $user_id ) ) {
+			$user_id = (int) $user_id;
+		}
+
+		if ( is_user_disabled( $user_id ) ) {
+			return new WP_Error(
+				'activitypub_user_not_found',
+				\__( 'Actor not found', 'activitypub' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		switch ( $user_id ) {
+			case self::BLOG_USER_ID:
+				return new Blog();
+			case self::APPLICATION_USER_ID:
+				return new Application();
+			default:
+				return User::from_wp_user( $user_id );
+		}
+
+		return new WP_Error(
+			'activitypub_user_not_found',
+			\__( 'Actor not found', 'activitypub' ),
+			array( 'status' => 404 )
+		);
+	}
+
+	/**
+	 * Get the Actor by username.
+	 *
+	 * @param string $username The Actor-Name.
```

Comment:
```suggestion
	 * @param string $username Name of the Actor.
```
Considered `Actorname` but that is probably not a thing.

Reply:
```suggestion
	 * @param string $username The Actor name.
```
Or this?